assets/app/view/game/loans.rb

```diff
@@ -12,8 +12,10 @@ class Loans < Snabberb::Component
       def render
         actions = @game.round.actions_for(@corporation)
         children = []
+        step = @game.round.active_step
 
         if actions.include?('payoff_loan')
+          payoff_loan_button_text = step.respond_to?(:payoff_loan_button_text) ? step.payoff_loan_button_text : 'Payoff Loan'
```

Comment:
I think we should remove the `button` from the attribute/method. The step doesn't care how its presented..only what the text should be.

Reply:
Button still in the method name.

Reply:
`scrap_button_text` was poor naming. The idea is the "view" layer (where the html is) is separate from the "model" layer (where the logic is). The model layer shouldn't know anything about how the html is structured. In this case, the knowledge that the view layer implements paying off loans via a button isn't something it should know. Additionally, the model layer shouldn't have to change if the button was changed to something something else. All the model layer has to do is tell the view layer what phrase to use for paying off a loan. This is why `payoff_loan_text` is a better name because now the model layer isn't implying anything about the "view" layer. Reference to the model - view architecture used by the site: https://www.geeksforgeeks.org/mvc-design-pattern/.

Reply:
And ideally, you could use something like `"Payoff #{@game.corp_loans_text}"`, so you don't need to provide a name twice. `corp_loans_text` is plural, so some work would have to be done there like make it `corp_loan_name` and then pluralize where/when needed.

Reply:
This was actually pretty simple, since 1849 only allows one bond to be issued, no plural was needed. Done.

Reply:
For what it's worth though, if you search button_text, you'll find that used in a lot of instances on the site.